src/Microsoft.AspNetCore.Razor.Language/RazorCodeDocumentExtensions.cs

```diff
@@ -3,13 +3,61 @@
 
 using System;
 using System.Collections.Generic;
+using System.Linq;
 using Microsoft.AspNetCore.Razor.Language.Intermediate;
 
 namespace Microsoft.AspNetCore.Razor.Language
 {
     public static class RazorCodeDocumentExtensions
     {
-        private static object TagHelperPrefixKey = new object();
+        // Internal for testing
+        internal static readonly string IdentifierKey = "identifier";
+        internal static readonly string ImportIdentifiersKey = "imports-identifiers";
+
+        public static string GetIdentifier(this RazorCodeDocument document)
```

Comment:
There's a big decision to make here about whether or not this should be part of the source document. After finishing this PR, I feel like it should be part of the source document. I'm planning to give that to @ajaybhargavb to play with. Thoughts?

Reply:
The identifier is whatever you want. It's a way for the project and runtime to communicate. By default this will be a relative path.